crates/extensions/tedge_log_manager/src/actor.rs

```diff
@@ -222,7 +222,13 @@ impl LogManagerActor {
         let path = match event {
             FsWatchEvent::Modified(path) => path,
             FsWatchEvent::FileDeleted(path) => path,
-            FsWatchEvent::FileCreated(path) => path,
+            // if we created the file earlier and now modify it, keep only later modify. some
+            // editors (e.g. vim) write to files by creating a temporary file, deleting old, and
+            // renaming temporary to previously deleted file. That way when we update, only 1 event
+            // is emitted.
+            //
+            // https://github.com/notify-rs/notify/issues/113#issuecomment-281836995
+            FsWatchEvent::FileCreated(_) => return Ok(()),
```

Comment:
There is an issue with that. If the log plugin starts while its configuration is *not* present, then it starts assuming a default configuration. But then, as file creation events are ignored, there is no way to provide a log plugin configuration without restarting the plugin. I agree this is a corner case.

Reply:
`FileCreated` means that a new file was created, but it still needs to be written to, so `FileCreated` is immediately followed by `FileModified` _most_ of the time, which the plugin picks up and correctly reloads.

I tested additional cases which result in successful reload:

- when the file is moved out and into the `/etc/tedge/plugins` directory (regardless if it was present when the plugin started) (the `FileCreated` event is not even emitted, because it's a `Rename` event, but `Modified` is still emitted)
- when plugin starts without the file and new one is created using `neovim` and by the shell (`echo "files = []" > tedge-log-plugin.toml`)
- when file is removed using `rm`

For all these the underlying library emits `EventKind::Modify(_)`.